src/NonBlocking/ConcurrentDictionary/DictionaryImpl`3.cs

```diff
@@ -170,12 +170,12 @@ private static uint CurrentTickMillis()
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         protected virtual int hash(TKey key)
         {
-            Debug.Assert(!(key is null));
+            Debug.Assert(key is not null);
 
             int h = _keyComparer.GetHashCode(key);
 
             // ensure that hash never matches TOMBPRIMEHASH, ZEROHASH, SPECIAL_HASH_BITS or 0
-            return h | (SPECIAL_HASH_BITS | (1 << 29));
```

Comment:
The goal is to make sure that this is a `variable | constant` without guessing if JIT is smart enough to do the same. 

This is one of the hottest places when the perf is not gated by cache misses. Perhaps it is better to keep it as it was - to be sure we are not getting `(h | SPECIAL_HASH_BITS)  |  (1 << 29)`  - with an extra `|` here.

Reply:
Sure, I'll revert it - the fix was applying the Roslyn auto-fixer on a suggestion to remove unnecessary braces.